broker/httpd/api.go

```diff
@@ -182,43 +202,35 @@ func StartServer(staticConfig *staticconfiguration.StaticConfiguration,
 			logger.Fatalf("Failed to start status server, err=%s", err)
 		}
 	}()
+	return server, nil
+}
 
-	var clientCACertPool *x509.CertPool
-	if len(staticConfig.Base.ClientCAFilename) > 0 {
-		clientCACertPool = x509.NewCertPool()
-		caCert, err := ioutil.ReadFile(staticConfig.Base.ClientCAFilename)
-		if err != nil {
-			logger.Fatalf("cannot read clientCA file err=%s", err)
-		}
-		clientCACertPool.AppendCertsFromPEM(caCert)
-	}
-
-	tlsConfig := &tls.Config{
-		MinVersion:               tls.VersionTLS12,
-		CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
-		PreferServerCipherSuites: true,
-		CipherSuites: []uint16{
-			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
-			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
-			tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
-		},
-		ClientAuth: tls.VerifyClientCertIfGiven,
-		ClientCAs:  clientCACertPool,
+func (s *Server) StartServicePort() error {
+	serviceListener, err := net.Listen("tcp", fmt.Sprintf(":%d", s.staticConfig.Base.ServicePort))
+	if err != nil {
+		return err
 	}
+	l := httpLogger{AccessLogger: s.accessLogger}
 	serviceServer := &http.Server{
-		Handler:      NewLoggingHandler(serviceMux, l),
-		TLSConfig:    tlsConfig,
+		Handler:      NewLoggingHandler(s.serviceMux, l),
+		TLSConfig:    s.tlsConfig,
 		ReadTimeout:  5 * time.Second,
 		WriteTimeout: 10 * time.Second,
 		IdleTimeout:  120 * time.Second,
 	}
+	c1 := make(chan error, 1)
 	go func() {
-		err := serviceServer.ServeTLS(serviceListener, staticConfig.Base.TLSCertFilename, staticConfig.Base.TLSKeyFilename)
-		if err != nil {
-			logger.Fatalf("Failed to start service server, err=%s", err)
-		}
```

Comment:
Please restore this conditional block. If it takes longer than 500ms and fails, there is no logging (silent failure).